core/src/main/java/com/linecorp/armeria/client/HttpClientDelegate.java

```diff
@@ -159,6 +165,20 @@ private void acquireConnectionAndExecute(ClientRequestContext ctx, Endpoint endp
         }
     }
 
+    private ProxyConfig selectProxyConfig(SessionProtocol sessionProtocol, String host, int port) {
+        ProxyConfig proxyConfig;
+        try {
+            final Endpoint endpoint = Endpoint.of(host, port);
+            proxyConfig = factory.proxyConfigSelector().select(sessionProtocol, endpoint);
```

Comment:
Let's just return here.
And we should use `handleEarlyRequestException()` if the `proxyConfig` is null instead of throwing an NPE.

Reply:
Ah, I forgot that the exception is caught by the below line. 😅 
So I guess it's okay then. 😄

Reply:
Sorry reviewers

This comment actually made me realize the API might be cleaner if we just throw as @minwoox suggested.
As a user, I don't think I would want random direct requests.

Let me work on a quick fix to just fail immediately in a separate revertable commit -- meanwhile any feedback is appreciated

Reply:
Pushed a small commit: https://github.com/line/armeria/pull/2752/commits/f5c687e6187e1920b2d9c136d1dbf166a547e1e4
Do let me know if you guys prefer the previous behavior though..

Reply:
Still LGTM

Reply:
👍 👍